tests/unit_tests/utils/landau_sampling.cpp

```diff
@@ -0,0 +1,86 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "detray/simulation/landau_distribution.hpp"
+
+// GTest include(s)
+#include <gtest/gtest.h>
+
+// System include(s)
+#include <algorithm>
+#include <iterator>
+
+using namespace detray;
+
+/// Test the landau distribution
+/// Validate that the mpv = -0.22278 for mu = 0 and sigma=1
+
+// Test class for covariance transport
+template <typename T>
+class LandauSamplingValidation : public ::testing::Test {
+    public:
+    using scalar_type = T;
+
+    // Function for getting the corresponding index of a value
+    std::size_t get_index(const scalar_type value) {
+        return static_cast<std::size_t>((value - min) / bin_size);
+    }
+
+    // Landau distribution with (mu = 0, sigma = 1) has the most probable value
+    // of -0.22278
+    constexpr static const scalar_type mu = 0.f;
+    constexpr static const scalar_type sigma = 1.f;
+    constexpr static const scalar_type mpv = -0.22278f;
+
+    // Binning information for counting
+    constexpr static const scalar_type bin_size = 0.05f;
+    constexpr static const scalar_type min = -2.f;
+    constexpr static const scalar_type max = 2.f;
+    constexpr static const std::size_t n_bins =
+        std::size_t((max - min) / bin_size);
+};
+
+// Test for float and double types
+using TestTypes = ::testing::Types<float, double>;
```

Comment:
Are we not running tests for double and float already via the scalar type? Why does it need to be done explicitly here?

Reply:
Right. Do you know how to bring `scalar` btw? I gave up after some trials

Reply:
Can we just go ahead